chain-impl-mockchain/src/fee.rs

```diff
@@ -0,0 +1,68 @@
+use crate::transaction as tx;
+use crate::value::Value;
+use chain_addr::Address;
+use std::ops::{Add, Mul};
+
+/// Linear fee using the basic affine formula `A * bytes(tx) + CONSTANT`.
+#[derive(PartialEq, Eq, PartialOrd, Debug, Clone, Copy)]
+pub struct LinearFee {
+    pub constant: Milli,
+    pub coefficient: Milli,
+}
+
+impl LinearFee {
+    pub fn new(constant: Milli, coefficient: Milli) -> Self {
+        LinearFee {
+            constant,
+            coefficient,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Debug, Clone, Copy)]
+pub struct Milli(u64);
```

Comment:
Should we worry about overflows? Here we put `Value`, which is u64 multiplied by 1000 and store it as `u64`, which could cause overflow. Maybe we should error-guard all the operations like in `Value` itself?

Reply:
How about storing internal value as u128?